be/src/udf/java/java_native_method.cpp

```diff
@@ -0,0 +1,70 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "udf/java/java_native_method.h"
+
+#include "column/column.h"
+#include "column/column_helper.h"
+#include "column/column_visitor_adapter.h"
+#include "column/nullable_column.h"
+#include "column/vectorized_fwd.h"
+#include "gutil/casts.h"
+#include "runtime/primitive_type.h"
+
+namespace starrocks::vectorized {
+
+class getColumnAddrVistor : public ColumnVisitorAdapter<getColumnAddrVistor> {
+public:
+    getColumnAddrVistor(jlong* jarr) : ColumnVisitorAdapter(this), _jarr(jarr) {}
+
+    Status do_visit(const NullableColumn& column) {
+        const auto& null_data = column.immutable_null_column_data();
+        _jarr[_idx++] = reinterpret_cast<int64_t>(null_data.data());
+        return column.data_column()->accept(this);
+    }
+
+    Status do_visit(const BinaryColumn& column) {
+        _jarr[_idx++] = reinterpret_cast<int64_t>(column.get_offset().data());
+        _jarr[_idx++] = reinterpret_cast<int64_t>(column.get_bytes().data());
+        return Status::OK();
+    }
+
+    template <typename T>
+    Status do_visit(const vectorized::FixedLengthColumn<T>& column) {
+        _jarr[_idx++] = reinterpret_cast<int64_t>(column.get_data().data());
+        return Status::OK();
+    }
+
+    template <typename T>
+    Status do_visit(const T& column) {
+        return Status::NotSupported("UDF Not Support Type");
+    }
+
+private:
+    size_t _idx = 0;
+    jlong* _jarr = 0;
+};
+
+jlong JavaNativeMethods::resizeStringData(JNIEnv* env, jclass clazz, jlong columnAddr, jint byteSize) {
+    Column* column = reinterpret_cast<Column*>(columnAddr);
+    BinaryColumn* binary_column = nullptr;
+    if (column->is_nullable()) {
+        binary_column = ColumnHelper::cast_to_raw<TYPE_VARCHAR>(down_cast<NullableColumn*>(column)->data_column());
+    } else {
+        binary_column = down_cast<BinaryColumn*>(column);
+    }
+    binary_column->get_bytes().resize(byteSize);
```

Comment:
Should we catch std::bad_alloc exception? what will happen in jvm if we don't do this?

Reply:
OK I'll test it

Reply:
```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
*** Aborted at 1652150487 (unix time) try "date -d @1652150487" if you are using GNU date ***
PC: @     0x7f344ad2c387 __GI_raise
*** SIGABRT (@0x3f30002425d) received by PID 148061 (TID 0x7f333f136700) from PID 148061; stack trace: ***
    @          0xae6a1d2 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7f344b7e1630 (unknown)
    @     0x7f344ad2c387 __GI_raise
    @     0x7f344ad2da78 __GI_abort
    @          0x4789ea4 _ZN9__gnu_cxx27__verbose_terminate_handlerEv.cold
    @          0xc933b56 __cxxabiv1::__terminate()
    @          0xc933ba1 std::terminate()
    @          0xc933cd4 __cxa_throw
    @          0x9d6bd83 starrocks::vectorized::JavaNativeMethods::resizeStringData()
    @     0x7f3236ccbb0e (unknown)
```

It seems JVM won't catch it, so I will try to catch it and throw a OOM Exception in Java

Reply:
```
mysql> select udf_int_cat(1,2);
ERROR 1064 (HY000): java.lang.OutOfMemoryError: OOM try to allocate 2 in java native function[com.starrocks.udf.UDFHelper.resizeStringData(Native Method), com.starrocks.udf.UDFHelper.getStringBoxedResult(UDFHelper.java:211), com.starrocks.udf.UDFHelper.getResultFromBoxedArray(UDFHelper.java:251)]
```